bpf/bpf_xdp.c

```diff
@@ -107,6 +107,7 @@ static __always_inline int check_v4(struct xdp_md *xdp)
 #ifdef CIDR4_FILTER
 	__builtin_memcpy(pfx.lpm.data, &ipv4_hdr->saddr, sizeof(pfx.addr));
 	pfx.lpm.prefixlen = 32;
+	pfx.pad = 0;
 
```

Comment:
I think here may be a bug: in the `lpm_v{4,6}_key` case you also add this in front of the key, therefore prefixlen needs to grow as well by as in the other lpm case due to the extra zeroed prefix.